internal/flowrate/util.go

```diff
@@ -7,24 +7,50 @@ package flowrate
 import (
 	"math"
 	"strconv"
+	"sync/atomic"
 	"time"
 )
 
 // clockRate is the resolution and precision of clock().
 const clockRate = 20 * time.Millisecond
 
-// czero is the process start time rounded down to the nearest clockRate
-// increment.
-var czero = time.Now().Round(clockRate)
+var (
+	hasInitializedClock = atomic.Bool{}
+	currentClockValue   = atomic.Int64{}
+	clockStartTime      = time.Time{}
+)
+
+// checks if the clock update timer is running. If not, sets clockStartTime and starts it.
+func ensureClockRunning() {
+	firstRun := hasInitializedClock.CompareAndSwap(false, true)
+	if !firstRun {
+		return
+	}
+	clockStartTime = time.Now().Round(clockRate)
+	go runClockUpdates()
+}
+
+// increments the current clock value every clockRate interval.
+func runClockUpdates() {
+	// Create a ticker that sends the current time on the channel every clockRate interval
+	ticker := time.Tick(clockRate)
+
+	// First tick happens after clockrate, therefore initial value is 0.
+	for range ticker {
```

Comment:
@cason @ValarDragon should we use time from the ticker? Because ticks might be dropped or delayed...

Reply:
like 

```go
for t := <-ticker {
  currentClockValue.Store(t.Round(clockRate))
}
```

Reply:
Your right, that is better! Updating, didn't realize the scheduler was unreliable at 20ms gaps